other/lru_cache.py

```diff
@@ -54,58 +60,70 @@ class Lru_Cache:
 
     >>> cache.set(3, 3)
 
-    >>> cache.get(2)
-    Traceback (most recent call last):
-    ...
-    ValueError: Key '2' not found in cache
+    >>> cache.get(2)    # None returned
 
     >>> cache.set(4, 4)
 
-    >>> cache.get(1)
-    Traceback (most recent call last):
-    ...
-    ValueError: Key '1' not found in cache
+    >>> cache.get(1)    # None returned
 
     >>> cache.get(3)
     3
 
     >>> cache.get(4)
     4
 
-    >>> cache.has_key(1)
-    False
+    >>> cache.cache_info()
+    'CacheInfo(hits=3, misses=2, capacity=2, current size=2)'
+
+    >>> @LruCache.decorator(100)
+    ... def fib(num):
+    ...     if num in (1, 2):
+    ...         return 1
+    ...     return fib(num - 1) + fib(num - 2)
 
-    >>> cache.has_key(4)
-    True
+    >>> for i in range(1, 100):
+    ...     res = fib(i)
+
+    >>> fib.cache_info()
+    'CacheInfo(hits=194, misses=99, capacity=100, current size=99)'
     '''
 
-    def __init__(self, capacity):
-        self.list = Double_Linked_List()
+    # class variable to map the decorator functions to their respective instance
+    decorator_function_to_instance_map = {}
+
+    def __init__(self, capacity: int):
+        self.list = DoubleLinkedList()
         self.capacity = capacity
         self.num_keys = 0
+        self.hits = 0
+        self.miss = 0
         self.cache = {}
```

Comment:
Why do we create both a cache and a decorator_function_to_instance_map?  Could we have one instead of two?

Reply:
We need both as the cache maps the keys to the Double Linked List Node